dspace-api/src/test/java/org/dspace/app/bulkedit/MetadataExportSearchIT.java

```diff
@@ -0,0 +1,226 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import com.google.common.io.Files;
+import com.opencsv.CSVReader;
+import com.opencsv.exceptions.CsvException;
+import org.dspace.AbstractIntegrationTestWithDatabase;
+import org.dspace.app.launcher.ScriptLauncher;
+import org.dspace.app.scripts.handler.impl.TestDSpaceRunnableHandler;
+import org.dspace.builder.CollectionBuilder;
+import org.dspace.builder.CommunityBuilder;
+import org.dspace.builder.ItemBuilder;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.Item;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.junit.Before;
+import org.junit.Test;
+
+public class MetadataExportSearchIT extends AbstractIntegrationTestWithDatabase {
+
+    private String subject1 = "subject1";
+    private String subject2 = "subject2";
+    private int numberItemsSubject1 = 30;
+    private int numberItemsSubject2 = 2;
+    private Item[] itemsSubject1 = new Item[numberItemsSubject1];
+    private Item[] itemsSubject2 = new Item[numberItemsSubject2];
+    private String filename;
+    private Collection collection;
+    TestDSpaceRunnableHandler testDSpaceRunnableHandler = new TestDSpaceRunnableHandler();
+    private ConfigurationService configurationService = DSpaceServicesFactory.getInstance().getConfigurationService();
+
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+
+        context.turnOffAuthorisationSystem();
+        Community community = CommunityBuilder.createCommunity(context).build();
+        collection = CollectionBuilder.createCollection(context, community).build();
+        String dspaceDir = configurationService.getProperty("dspace.dir");
+        filename = String.format("%s%smetadataExportSearch.csv", dspaceDir, File.separator);
+
+
+        for (int i = 0; i < numberItemsSubject1; i++) {
+            itemsSubject1[i] = ItemBuilder.createItem(context, collection)
+                .withTitle(String.format("%s item %d", subject1, i))
+                .withSubject(subject1)
+                .withIssueDate("2020-09-" + i)
+                .build();
+        }
+
+        for (int i = 0; i < numberItemsSubject2; i++) {
+            itemsSubject2[i] = ItemBuilder.createItem(context, collection)
+                .withTitle(String.format("%s item %d", subject2, i))
+                .withSubject(subject2)
+                .withIssueDate("2021-09-" + i)
+                .build();
+        }
+        context.restoreAuthSystemState();
+    }
+
+    private void checkItemsPresentInFile(String filename, Item[] items) throws IOException, CsvException {
```

Comment:
as by previous comment about sorting, the order check can be postponed but presence of no duplicate should be included (it would be enough to just check that `items.length == lines.size() -1`